data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/DefaultParameterExpression.java

```diff
@@ -48,6 +51,13 @@ public QueryParameterBinding bind(BindingContext bindingContext) {
         if (outgoingQueryParameterProperty == null) {
             return new SimpleParameterBinding(name, DataType.forType(paramClass), bindingContext.isExpandable(), value);
         }
-        return new PropertyPathParameterBinding(name, outgoingQueryParameterProperty, bindingContext.isExpandable(), value);
+        Object parameterValue = value;
+        if (value != null && outgoingQueryParameterProperty.getProperty().getOwner().isEmbeddable()) {
```

Comment:
@dstepanov So here is the case that `DefaultParameterExpression` gets value as embeded id but binding parameters should be each field of embedded id. So, we need to extract value for the param if field belongs to embedded id and passed value is embedded id. Not sure if this is safe enough and if there is some better approach to extract correct value for embedded id part(s).

Reply:
I think it should be expanded in the equals operator, similar to the composite identity:

https://github.com/micronaut-projects/micronaut-data/blob/4.10.x/data-model/src/main/java/io/micronaut/data/model/query/builder/sql/AbstractSqlLikeQueryBuilder2.java#L2019-L2032

Reply:
That should be done a bit differently:
- We need to add `PersistentPropertyPath parameterPropertyPath` to `DefaultParameterExpression` like it's in `SourceParameterExpressionImpl`
- When the property is expanded, we should try to recognize `DefaultParameterExpression` and clone it to a new parameter with `parameterPropertyPath` set the left property path
- `DefaultParameterExpression` should use the same logic as in the source one to calculate `parameterBindingPath`
- `DefaultBindableParametersStoredQuery` should support using `getParameterBindingPath` for cases when the value set